packages/gatsby-link/src/index.js

```diff
@@ -1,12 +1,20 @@
 import PropTypes from "prop-types"
 import React from "react"
-import { Link } from "@reach/router"
+import { Link, Location } from "@reach/router"
+import { resolve } from "@reach/router/lib/utils"
 
 import { parsePath } from "./parse-path"
 
 export { parsePath }
 
+function isRelativePath(path) {
+  return /^\.{1,2}\/(?!\/)/.test(path)
+}
+
 export function withPrefix(path) {
+  if (isRelativePath(path)) {
+    normalizePath(path)
+  }
```

Comment:
is this doing anything? because you're not returning it.

Reply:
Oops!